tools/bots/mission-control-bot.ts

```diff
@@ -222,6 +269,13 @@ const botActionFaucetSend = async (msg: Message, authorId: string, messageConten
   receivers[authorId] = Date.now();
 
   try {
+    // push to TODO queue
+    pendingQueue.push(`${authorId}:0x${address}`);
+
+    // wait for our item to be first in the list
+    await waitForQueue(authorId, address);
```

Comment:
I agree that this may lead to race conditions in the future if the bot gets more and more complex. 

About the pendingQueueProcessor, would it submit all the transactions in the same block in batch? Because if not, we would need to sync the different executions of the processor to not use the same nonce

Reply:
The idea was to execute them all in the same batch yes.
Every 12000ms it wakes up, copy the pendingQueue, and execute all the tx in there and goes back to sleep